src/resty/aws/request/build.lua

```diff
@@ -73,16 +73,21 @@ local function parse_query(q)
 end
 
 local function get_host_port(config)
-  local host = config.endpoint or config.globalEndpoint
+  local endpoint = config.endpoint or config.globalEndpoint
   do
-    local s, e = host:find("://")
+    local s, e = endpoint:find("://")
+    local scheme = config.scheme
     if s then
-      -- the "globalSSL" one from the region_config_data file
-      local scheme = host:sub(1, s-1):lower()
-      host = host:sub(e+1, -1)
-      if config.tls == nil then
-        config.tls = scheme == "https"
-      end
+      scheme = scheme or endpoint:sub(1, s-1):lower()
+      endpoint = endpoint:sub(e+1, -1)
+    end
+
+    scheme = scheme or "https"
+    config.scheme = scheme
+
+    -- the "globalSSL" one from the region_config_data file
+    if config.tls == nil then
+      config.tls = scheme == "https"
```

Comment:
```suggestion
      -- the "globalSSL" one from the region_config_data file
      scheme = scheme or endpoint:sub(1, s-1):lower()
      endpoint = endpoint:sub(e+1, -1)
    end

    scheme = scheme or "https"
    config.scheme = scheme

    if config.tls == nil then
      config.tls = scheme == "https"
```

The comment is now in the wrong position

Reply:
done